src/django_components/component.py

```diff
@@ -1062,22 +1074,49 @@ def _render_impl(
                 new_output = self.on_render_after(context, template, html_content)
                 html_content = new_output if new_output is not None else html_content
 
-                output = postprocess_component_html(
-                    component_cls=self.__class__,
-                    component_id=self.id,
-                    html_content=html_content,
-                    css_input_hash=css_input_hash,
-                    js_input_hash=js_input_hash,
-                    type=type,
-                    render_dependencies=render_dependencies,
-                )
-
         # After rendering is done, remove the current state from the stack, which means
         # properties like `self.context` will no longer return the current state.
         self._render_stack.pop()
         context.render_context.pop()
 
-        return output
+        # Internal component HTML post-processing:
```

Comment:
This is part of the change to make it possible to parse the component HTML only once. It's achieved by deferring the postprocessing of the component output until its parent has been post-processed.

For context see ["Idea 2 - Render components top-down instead of bottom-up"](https://github.com/django-components/django-components/issues/14#issuecomment-2596096632)